cmd/kubeadm/app/componentconfigs/utils_test.go

```diff
@@ -0,0 +1,131 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package componentconfigs
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestUnsupportedConfigVersionError_Error(t *testing.T) {
+	type fields struct {
+		OldVersion     schema.GroupVersion
+		CurrentVersion schema.GroupVersion
+		Document       []byte
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   string
+	}{
+		{
+			name: "versions are valid",
+			fields: fields{
+				OldVersion: schema.GroupVersion{
+					Group:   "admissionregistration.k8s.io",
+					Version: "v1beta1",
+				},
+				CurrentVersion: schema.GroupVersion{
+					Group:   "admissionregistration.k8s.io",
+					Version: "v1",
+				},
+			},
+			want: "unsupported apiVersion \"admissionregistration.k8s.io/v1beta1\", you may have to do manual conversion to \"admissionregistration.k8s.io/v1\" and run kubeadm again",
+		},
+		{
+			name: "versions are empty",
+			fields: fields{
+				OldVersion:     schema.GroupVersion{},
+				CurrentVersion: schema.GroupVersion{},
+			},
+			want: "unsupported apiVersion \"\", you may have to do manual conversion to \"\" and run kubeadm again",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := &UnsupportedConfigVersionError{
+				OldVersion:     tt.fields.OldVersion,
+				CurrentVersion: tt.fields.CurrentVersion,
+				Document:       tt.fields.Document,
+			}
+			if got := err.Error(); got != tt.want {
+				t.Errorf("UnsupportedConfigVersionError.Error() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestUnsupportedConfigVersionsErrorMap_Error(t *testing.T) {
+	tests := []struct {
+		name string
+		errs UnsupportedConfigVersionsErrorMap
+		want string
+	}{
+		{
+			name: "only one error",
+			errs: UnsupportedConfigVersionsErrorMap{
+				"foo": &UnsupportedConfigVersionError{
+					OldVersion: schema.GroupVersion{
+						Group:   "admissionregistration.k8s.io",
+						Version: "v1beta1",
+					},
+					CurrentVersion: schema.GroupVersion{
+						Group:   "admissionregistration.k8s.io",
+						Version: "v1",
+					},
+				},
+			},
+			want: "multiple unsupported config version errors encountered:\n\t- unsupported apiVersion \"admissionregistration.k8s.io/v1beta1\", you may have to do manual conversion to \"admissionregistration.k8s.io/v1\" and run kubeadm again",
+		},
+		{
+			name: "multiple errors",
+			errs: UnsupportedConfigVersionsErrorMap{
+				"err1": &UnsupportedConfigVersionError{
```

Comment:
given an ErrorMap includes an Error can we just do all the testing in this function and remove the other test function?

this way we avoid duplicate testing.